### PR TITLE
Initial Code: Added kirtify and replace functions to parse template syntax

### PR DESCRIPTION
**Pull request by Vector919**
Added functionality for parsing template matches with regex objects

### CODE REVIEW

File: src/proact/core.cljs [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 
-(enable-console-print!)
+(defn replace_match [match context]
+    ;; eval a match and then if it works return it
+    (def res (.replace match (get (keyword (.split match #"\.") ) context)))
+    res
+)
 
-(println "Hello world!")
+(defn kirtify [vars context]
```
Comment (agamdua) [inline src/proact/core.cljs]: lets change the name now :)

File: src/proact/core.cljs [hunk 2]
```diff
@@ -1,8 +1,17 @@
 (ns proact.core
   (:require [clojure.browser.repl :as repl]))
-
 ;; (repl/connect "http://localhost:9000/repl")
 
-(enable-console-print!)
+(defn replace_match [match context]
```
Comment (agamdua) [inline src/proact/core.cljs]: https://github.com/bbatsov/clojure-style-guide#naming

File: src/proact/core.cljs [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+    (def res (.replace match (get (keyword (.split match #"\.") ) context)))
+    res
+)
 
-(println "Hello world!")
+(defn kirtify [vars context]
+    (def parse_vars (js/RegExp "{{([^}}]+)}}"))
+    (def matches (.exec parse_vars vars))
+    (def line (nth matches 1))
+    (def final_vars (replace_match line context))
```
Comment (agamdua) [inline src/proact/core.cljs]: https://github.com/bbatsov/clojure-style-guide#naming

File: src/proact/core.cljs [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
-(enable-console-print!)
+(defn replace_match [match context]
+    ;; eval a match and then if it works return it
+    (def res (.replace match (get (keyword (.split match #"\.") ) context)))
+    res
+)
 
-(println "Hello world!")
+(defn kirtify [vars context]
+    (def parse_vars (js/RegExp "{{([^}}]+)}}"))
```
Comment (agamdua) [inline src/proact/core.cljs:12]: Do want to use this as a constant in the file?

(so we can "isolate" it from the code)

Comment (Vector919) [inline src/proact/core.cljs:12]: not making a constants.py if that's what you want

Comment (agamdua) [inline src/proact/core.cljs:12]: A #define macro would be sufficient

**Comment by Vector919** [comment]
@agamdua uhh. you sure there's `#define` macro's in this language? (ClojureScript specifially)

take a look @ https://github.com/clojure/clojurescript/wiki/Differences-from-Clojure

File: src/proact/core.cljs (hunk 1 above)
Comment (Vector919) [inline src/proact/core.cljs]: @agamdua noted, although be aware I see this as an attempt to devalue my contributions to the greatest templating engine in existence

**Comment by agamdua** [comment]
@Vector919 C has `#define`. We need a C extension.

Doesn't the JVM execute for ClojureScript.

Boom.

**Pull request merged**